app/models/enterprise_group.rb

```diff
@@ -28,4 +39,50 @@ class EnterpriseGroup < ActiveRecord::Base
 
   scope :by_position, order('position ASC')
   scope :on_front_page, where(on_front_page: true)
+  scope :managed_by, lambda { |user|
+    if user.has_spree_role?('admin')
+      scoped
+    else
+      where('owner_id = ?', user.id);
+    end
+  }
+
+  def set_unused_address_fields
+    address.firstname = address.lastname = 'unused' if address.present?
+  end
+
+  def set_undefined_address_fields
+    if !address.present?
+      return
+    end
+    address.phone.present? || address.phone = 'undefined'
```

Comment:
I think this is almost exactly equivalent and a slightly more concise way of expressing this:

```ruby
address.phone ||= 'undefined'
```